modtools/dynamic-adv-reactions.lua

```diff
@@ -0,0 +1,882 @@
+-- Tool to allow dynamically enabling / disabling adventure mode reactions based on circumstances
+--@ module = true
+-- Created for dfhack version 0.44.12-r3
+
+--[[ HOW IT WORKS:
+Commands / scripts register a set of conditional checks associated with a reaction (the set of conditions is referred to a "condition blob").
+When the player opens up the actions menu in game (the menu where they can perform reactions), the script checks if they satisfy one of the condition blobs associated with that reaction (that is, they satisfy every condition in any of the registered blobs for that reaction).
+If they don't satisfy all the conditions for any of the condition blobs, the reaction is removed from their list of reactions.
+]]
+
+--[[ TODO / WISHLIST:
+- Time: Available within certain hourly bounds, allowing for day/night requirements.
+- Season. Available during a particular season.
+- Weather. Available during certain weathers.
+]]
+
+local help = [====[
+
+modtools/dynamic-adv-reactions
+==============================
+Tool to allow dynamically enabling / disabling adventure mode reactions based on certain conditions.
+When a reaction is registered with this script, its usage by adventurers will be disabled by default. Each usage of this command creates its own set of conditions to check, and the reaction will be enabled for the player if they satisfy all the criteria of one of those sets of conditions.
+Registered reactions are cleared when the world is unloaded, and so it is best to include any commands for this within an `onLoad*.init` file.
```

Comment:
Inline code blocks need double backticks (single backticks make links):
```suggestion
Registered reactions are cleared when the world is unloaded, and so it is best to include any commands for this within an ``onLoad*.init`` file.
```